segment/writer.go

```diff
@@ -367,15 +367,46 @@ func (w *Writer) appendEntry(e types.LogEntry) error {
 		typ: FrameEntry,
 		len: uint32(len(e.Data)),
 	}
-	return w.appendFrame(fh, e.Data)
+
+	bufOffset, err := w.appendFrame(fh, e.Data)
+	if err != nil {
+		return err
+	}
+
+	// If frame is an entry, update the index
+	offsets := w.getOffsets()
+
+	if len(offsets) == 0 && w.info.BaseIndex == 1 {
+		for i := 0; i < int(e.Index)-1; i++ {
+			offsets = append(offsets, 0)
+		}
+
+		atomic.StoreUint64(&w.info.MinIndex, e.Index)
+	}
+
+	// Add the index entry. Note this is safe despite mutating the same backing
+	// array as tail because it's beyond the limit current readers will access
+	// until we do the atomic update below. Even if append re-allocates the
+	// backing array, it will only read the indexes smaller than numEntries from
+	// the old array to copy them into the new one and we are not mutating the
+	// same memory locations. Old readers might still be looking at the old
+	// array (lower than numEntries) through the current tail.offsets slice but
+	// we are not touching that at least below numEntries.
```

Comment:
Should we move this comment above the  `if` since this commentary on why it's safe to append to the slice without a lock is just as true in the loop above as here?